Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp

```diff
@@ -44,14 +45,18 @@ void MolDraw2DCairo::initDrawing() {
     initTextDrawer(df_noFreetype);
   } else {
 #ifdef RDK_BUILD_FREETYPE_SUPPORT
+    // A dynamic cast requires a public interface which DrawText doesn't
+    // have.  But this should always be right.
     if (df_noFreetype) {
-      dynamic_cast<DrawTextCairo *>(text_drawer_.get())->setCairoContext(dp_cr);
-    } else {
-      dynamic_cast<DrawTextFTCairo *>(text_drawer_.get())
+      static_cast<MolDraw2D_detail::DrawTextCairo *>(text_drawer_.get())
           ->setCairoContext(dp_cr);
+    } else {
+      MolDraw2D_detail::DrawTextFTCairo *dt =
+          static_cast<MolDraw2D_detail::DrawTextFTCairo *>(text_drawer_.get());
+      dt->setCairoContext(dp_cr);
     }
 #else
-    dynamic_cast<DrawTextCairo *>(text_drawer_.get())->setCairoContext(dp_cr);
+    static_cast<DrawTextCairo *>(text_drawer_.get())->setCairoContext(dp_cr);
```

Comment:
IIRC, this is from the old iteration when all of the DrawText classes had private inheritance including private constructors.  In that case, the dynamic_cast compiles ok but returns nullptr at runtime because the cast fails.  It took some hunting to discover this was the cause the sudden runtime failures in the test code.  I will try reverting to dynamic_cast with the current fully public interface.